src/main/java/com/minelittlepony/bigpony/mod/mixin/MixinRenderLivingBase.java

```diff
@@ -25,7 +25,9 @@ protected MixinRenderLivingBase(RenderManager renderManager) {
             at = @At(value = "INVOKE",
                     target = "Lnet/minecraft/client/renderer/GlStateManager;scale(FFF)V",
                     shift = Shift.AFTER))
-    private void onPrepareScale(EntityLivingBase entity, float ticks, CallbackInfoReturnable ci) {
-        LiteLoader.getInstance().getMod(LiteModBigPony.class).onRenderEntity(entity);
+    private void onPrepareScale(EntityLivingBase entity, float ticks, CallbackInfoReturnable<Float> ci) {
+        LiteModBigPony bp = LiteLoader.getInstance().getMod(LiteModBigPony.class);
+        bp.onRenderEntity(entity);
+        this.shadowSize = bp.getUpdatedShadowSize(this.shadowSize, entity);
```

Comment:
Try not to save individual properties as global. It makes it harder when multiple players are around. This is especially bad because you never set it back to what it was before. If you don't set it back, foals will have adult-sized shadows. There's lots of other problems that could sprout up from this as well.

A better solution would be to inject the shadow rendering code in `Render` and make it smaller there.

Reply:
This was one of the causes of a couple bugs way back with minelittlepony.

Reply:
Hm... Good point. Right now it will simply reset the value to an appropriate value every time a player is rendered, so if there's anyone else trying to alter the shadow size their preference will be overriden.

Last I checked, MineLP didn't try to do this, though. I'm also not fond of the idea to duplicate the shadow rendering code, otherwise you'd likely get to the same type of thing I've seen with MineLP before where the shadows are twice as dark as they should be.